### PR TITLE
thrift 0.12.0

### PR DESCRIPTION
**Pull request by taraspos**
- ✅Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- ✅ Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- ✅ Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- ✅ Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- ✅Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by fxcoudert** [comment]
- Please squash both commits into a single one, `thrift 0.12.0`
- Add to this commit a rename of `Aliases/thrift@0.11` to `Aliases/thrift@0.12`
- In a second commit, bump revision of `osquery`

**Comment by Trane9991** [comment]
@fxcoudert 
thanks for the feedback, will do.
But I didn't get the one about `osquery`.
Do you mean update https://github.com/Homebrew/homebrew-core/blob/master/Formula/osquery.rb to 3.3.1?

**Comment by Trane9991** [comment]
About the `osquery`, it is not published on the official downloads page yet https://osquery.io/downloads/official, and I can't find the `sha256`for the package anywhere else.

**Comment by fxcoudert** [comment]
I mean in `osquery`, bump `revision 1` to `revision 2`, in a separate commit with title `osquery: revision for thrift`

**Comment by Trane9991** [comment]
All done

**Comment by fxcoudert** [comment]
Removing the seldom-used options, per #31510

```
==> install (30 days)
Index | Name (with options)                                   | Count |  Percent
-----:|-------------------------------------------------------|------:|--------:
1     | thrift                                                | 1,439 |   97.30%
2     | thrift --HEAD                                         |    21 |    1.42%
3     | thrift --with-java                                    |     7 |    0.47%
4     | thrift --with-python@2                                |     7 |    0.47%
```

**Comment by fxcoudert** [comment]
`osquery` build failure…

**Comment by Trane9991** [comment]
@fxcoudert any ideas what can be done here? 😕

File: Formula/thrift.rb [hunk 1]
```diff
@@ -36,9 +36,11 @@ def install
       --with-openssl=#{Formula["openssl"].opt_prefix}
       --without-erlang
       --without-haskell
+      --without-java
```
Comment (Fokko) [inline Formula/thrift.rb]: Why are you removing Java, I think it is important to have the Java bindings. If so, you need to remove the env-settings as well.

File: Formula/thrift.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 class Thrift < Formula
   desc "Framework for scalable cross-language services development"
   homepage "https://thrift.apache.org/"
-  url "https://www.apache.org/dyn/closer.cgi?path=/thrift/0.11.0/thrift-0.11.0.tar.gz"
-  sha256 "c4ad38b6cb4a3498310d405a91fef37b9a8e79a50cd0968148ee2524d2fa60c2"
+  url "https://www.apache.org/dyn/closer.cgi?path=/thrift/0.12.0/thrift-0.12.0.tar.gz"
```
Comment (Fokko) [inline Formula/thrift.rb]: This url isn't working for me:
```
MacBook-Pro-van-Fokko:parquet-mr fokkodriesprong$ curl -o thrift-0.12.0.tar.gz -L  https://www.apache.org/dyn/closer.cgi?path=/thrift/0.12.0/thrift-0.12.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 18711    0 18711    0     0  77732      0 --:--:-- --:--:-- --:--:-- 77639
MacBook-Pro-van-Fokko:parquet-mr fokkodriesprong$ ls -lah thrift-0.12.0.tar.gz 
-rw-r--r--  1 fokkodriesprong  staff    18K 24 jan 20:01 thrift-0.12.0.tar.gz
MacBook-Pro-van-Fokko:parquet-mr fokkodriesprong$ head thrift-0.12.0.tar.gz 
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="description" content="Home page of The Apache Software Foundation">
  <base href="https://www.apache.org/dyn/closer.cgi" />
  <link rel="apple-touch-icon" sizes="57x57" href="/favicons/apple-touch-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/favicons/apple-touch-icon-60x60.png">
```

File: Formula/thrift.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/thrift.rb]: Barely anyone uses it based on the analytics above and it pulls in a huge dependency, I'd say this is a valid reason to remove it.

File: Formula/thrift.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/thrift.rb]: I think it depends on the user agent and other settings, it's used everywhere throughout homebrew.

Comment (Trane9991) [reply Formula/thrift.rb]: Yeah, this link redirects to the closest mirror

**Comment by Fokko** [comment]
Any update on this?

**Comment by adamlc** [comment]
Would love an update if there is one 👍

**Comment by linguofeng** [comment]
Any updates?

**Comment by Trane9991** [comment]
Sorry guys, I have no idea what I can do here. The most I can try is to rerun the build, maybe it will work now ¯\_(ツ)_/¯.

The upstream issue seems to be ignored.

**Comment by fxcoudert** [comment]
The `osquery` build should be fixed with the new version: 8b42b60bf3ab0775d95b844e15b75d214d26d07f

Can you bump its revision now? That should do the trick

**Comment by fxcoudert** [comment]
Thanks @Trane9991 ! 🍸

**Pull request closed without merging**